### DISCUSSION TITLE
I2C1, STM32F746 does not work with Wire Library

### DISCUSSION THREAD

**Discussion by epikao** [stm32duino/Arduino_Core_STM32#1565] [Libraries] [answered]
Hello, I tried everything, but the I2C1 ( PB8 -> I2C1_SCL  and PB09 -> I2C1_SDA ) with the Wire library does not work. 
Measured with a logic analyzer the I2C lines are always high.

What's wrong :-( ?

See code below:

```
TwoWire Wire2(PB9, PB8);

void setup();
{
  pinMode(PB9, INPUT);  // remove internal pullup
  pinMode(PB8, INPUT);  // remove internal pullup
  Wire2.begin();
}

void loop()
{
          Wire2.beginTransmission(0x38);
          Wire2.write(0x00);
          Wire2.endTransmission();
          delay(20);
          Wire2.requestFrom(0x38, 30);

          for(uint8_t i=0;i<=15 && Wire2.available(); i++){
              buffer[i]= Wire2.read();
              Serial6.println(buffer[i]);
              
          }
}
```

Thank you

**Comment by ABOSTM** [comment]
Hi @epikao,
You initially configure PB9/PB8 as I2C (Alternate function), with
`TwoWire Wire2(PB9, PB8);`
Fine, but in then in `setup()` you configure those pins as input so implicitly they are configured as GPIO (and no more configured as I2C)
  ```
pinMode(PB9, INPUT);  // remove internal pullup
  pinMode(PB8, INPUT);  // remove internal pullup
```
By the way, by default, I2C pins are configured without internal pullup (thus requiring external pullup) so you don't need to remove internal pull up.

So just remove those 2 lines and I hope that would be good.
  ```
pinMode(PB9, INPUT);  // remove internal pullup
  pinMode(PB8, INPUT);  // remove internal pullup
```

**Reply by epikao** [reply]
> So just remove those 2 lines and I hope that would be good.
> 
> ```
> pinMode(PB9, INPUT);  // remove internal pullup
> pinMode(PB8, INPUT);  // remove internal pullup
> ```


Thank you. Initially I did not include these two lines, but with or without, both i2c lines remain constantly high. No clock or data measurable.

**Comment by epikao** [comment, marked as answer]
Hello, I have found the error. Although the display manufacturer states PB8 and PB9 as the I2C port, this is not true, it is PH8 and PH7.
I do not understand, but it works...
